interpreter/attributes.go

```diff
@@ -860,5 +860,11 @@ func refResolve(adapter ref.TypeAdapter, idx ref.Val, obj interface{}) (ref.Val,
 		}
 		return elem, nil
 	}
+	if types.IsUnknown(celVal) {
+		return celVal, nil
+	}
+	if types.IsError(celVal) {
```

Comment:
I thought we were going to have the ability to bind variables to reified errors.  Does this interfere with that?

Reply:
The error message would be preserved if the variable is bound to an error, but if we add source positions as extra state carried with the `types.Err` value, then this would be lost. I'll add a `TODO` to make sure this is accounted for in the future. Currently, the error behavior is structured to be consistent with the rest of the code.